src/main/java/com/puppycrawl/tools/checkstyle/utils/UnmodifiableCollectionUtil.java

```diff
@@ -63,6 +65,21 @@ public static <T> List<T> unmodifiableList(List<T> collection) {
         return Collections.unmodifiableList(collection);
     }
 
+    /**
+     * Returns an unmodifiable view of a List containing elements of a specific type.
+     *
+     * @param items The List of items to make unmodifiable.
+     * @param elementType The Class object representing the type of elements in the list.
+     * @param <S> The generic type of elements in the input Collection.
+     * @param <T> The type of elements in the resulting unmodifiable List.
+     * @return An unmodifiable List containing elements of the specified type.
+     */
+    public static <S, T> List<T> unmodifiableList(Collection<S> items, Class<T> elementType) {
+        return items.stream()
+                .map(elementType::cast)
+                .collect(Collectors.toUnmodifiableList());
+    }
```

Comment:
Why did we make this about collections, when the issue is with the calls to `Class#cast`?  If we want to make some hack to bypass pitest, why not just have a util method to cast the stream as an intermediate operation?

Pseudocode:
```
<R, T> Stream<T> castedStreamOF(Stream<R> s, Class<T> c) {
  return stream.map(c::cast)
}

```

Then we can collect or perform subsequent stream operations elsewhere more readily.

Reply:
It has unmodified collection, that pitest will complain about, not sure why it is not doing it now.

And current implementation is easier, and we don't want to make it highly reusable, at least for now.

Reply:
Why is it a public method in a utility class if we don't want it to be reusable?

Reply:
Pitest limitation to allow easy exclude from mutation by class name